daq/runner.py

```diff
@@ -299,12 +317,14 @@ def main_loop(self):
                                                    loop_hook=self._loop_hook,
                                                    timeout_sec=20)  # Polling rate
             self.stream_monitor = monitor
-            self.monitor_stream('faucet', self.faucet_events.sock, self._handle_faucet_events)
+            self.monitor_stream('faucet', self.faucet_events.sock, self._handle_faucet_events,
+                                priority=10)
             if self.event_trigger:
                 self._flush_faucet_events()
             LOGGER.info('Entering main event loop.')
             LOGGER.info('See docs/troubleshooting.md if this blocks for more than a few minutes.')
             while self.stream_monitor.event_loop():
+                self._monitor_disconnected_port()
```

Comment:
Seems like this can/should be pushed down into the _module_heartbeat() function/system?  Why does this need to be its own top-level thing.  Seems like this (ip timeout) is a related case to the overall module timeout case... would like to keep them as similar as possible.

Reply:
Module time out should be at the host level and 1 test timing out should have no effect on the remaining tests. Port flap timeout should be at the runner level since the timeout for that should result in the killing of the host.